🔧  Missing namespace information required by Gradle 8

### How were you trying to build the app?

Tried to build my app using `pnpm run android`

### Full build logs

```tsx
> Task :react-native-vision-camera:processDebugManifest FAILED

Incorrect package="com.mrousavy.camera" found in source AndroidManifest.xml: [redacted]/node_modules/react-native-vision-camera/android/src/main/AndroidManifest.xml.
Setting the namespace via the package attribute in the source AndroidManifest.xml is no longer supported.
Recommendation: remove package="com.mrousavy.camera" from the source AndroidManifest.xml: [redacted]/node_modules/react-native-vision-camera/android/src/main/AndroidManifest.xml.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-vision-camera:processDebugManifest'.
> A failure occurred while executing com.android.build.gradle.tasks.ProcessLibraryManifest$ProcessLibWorkAction
   > Incorrect package="com.mrousavy.camera" found in source AndroidManifest.xml: [redacted]/node_modules/react-native-vision-camera/android/src/main/AndroidManifest.xml.
     Setting the namespace via the package attribute in the source AndroidManifest.xml is no longer supported.
     Recommendation: remove package="com.mrousavy.camera" from the source AndroidManifest.xml: [redacted]/node_modules/react-native-vision-camera/android/src/main/AndroidManifest.xml.
```


### Project dependencies

```json
"dependencies": {
    "@aws-amplify/auth": "^5.6.5",
    "@aws-amplify/core": "^5.8.5",
    "@aws-amplify/pushnotification": "^5.0.45",
    "@aws-amplify/rtn-push-notification": "^1.1.7",
    "@react-native-async-storage/async-storage": "^1.19.3",
    "@react-native-community/netinfo": "^9.4.1",
    "@react-native-masked-view/masked-view": "^0.3.0",
    "@react-navigation/native": "^6.1.9",
    "@react-navigation/stack": "^6.3.20",
    "@rescript/core": "^0.5.0",
    "@rescript/react": "^0.11.0",
    "@sentry/react-native": "^5.11.1",
    "@urql/rescript": "^4.0.0",
    "amazon-cognito-identity-js": "^6.3.6",
    "graphql": "^16.8.1",
    "jsc-android": "^250231.0.0",
    "react": "18.2.0",
    "react-intl": "^6.5.1",
    "react-native": "0.72.6",
    "react-native-gesture-handler": "^2.13.4",
    "react-native-get-random-values": "^1.9.0",
    "react-native-gradle-plugin": "^0.71.19",
    "react-native-localize": "^3.0.2",
    "react-native-reanimated": "^3.5.4",
    "react-native-safe-area-context": "^4.7.4",
    "react-native-screens": "^3.25.0",
    "react-native-url-polyfill": "^2.0.0",
    "react-native-vision-camera": "^3.6.4",
    "rescript-react-native": "^0.72.0",
    "rescript-react-navigation": "^6.0.3",
    "rn-slide-button": "^1.0.3",
    "urql": "^4.0.5",
    "zustand": "^4.4.4"
  },
```


### VisionCamera Version

3.6.4

### Target platforms

Android

### Operating system

MacOS

### Can you build the VisionCamera Example app?

I didn't try (⚠️ your issue might get ignored & closed if you don't try this)

### Additional information

- [ ] I am using Expo
- [ ] I have enabled Frame Processors (react-native-worklets-core)
- [X] I have read the [Troubleshooting Guide](https://react-native-vision-camera.com/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

--- a/package/android/build.gradle
+++ b/package/android/build.gradle
@@ -73,7 +73,7 @@ repositories {
 }
 
 android {
-  namespace "com.mrousavy.camera.example"
+  namespace "com.mrousavy.camera"
 
   // Used to override the NDK path/version on internal CI or by allowing
   // users to customize the NDK path/version from their root project (e.g. for M1 support)

--- a/package/android/build.gradle
+++ b/package/android/build.gradle
@@ -1,4 +1,8 @@
 import java.nio.file.Paths
+import com.android.Version
+
+def agpVersion = Version.ANDROID_GRADLE_PLUGIN_VERSION.tokenize('.')[0].toInteger()
+def androidManifestPath = agpVersion >= 7 ? 'src/main/AndroidManifest.xml' : 'src/hasNamespace/AndroidManifest.xml'
 
 buildscript {
   def kotlin_version = rootProject.ext.has("kotlinVersion") ? rootProject.ext.get("kotlinVersion") : project.properties["VisionCamera_kotlinVersion"]
@@ -73,7 +77,9 @@ repositories {
 }
 
 android {
-  namespace "com.mrousavy.camera"
+  if (agpVersion >= 7) {
+    namespace "com.mrousavy.camera"
+  }
 
   // Used to override the NDK path/version on internal CI or by allowing
   // users to customize the NDK path/version from their root project (e.g. for M1 support)
@@ -104,6 +110,12 @@ android {
                 "-DENABLE_FRAME_PROCESSORS=${hasWorklets}"
         abiFilters (*reactNativeArchitectures())
       }
+    }
+  }
+
+  sourceSets {
+    main {
+      manifest.srcFile androidManifestPath
     }
   }
 

--- a/package/android/src/hasNamespace/AndroidManifest.xml
+++ b/package/android/src/hasNamespace/AndroidManifest.xml
@@ -1,0 +1,4 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+          package="com.mrousavy.camera">
+
+</manifest>

--- a/package/android/src/main/AndroidManifest.xml
+++ b/package/android/src/main/AndroidManifest.xml
@@ -1,4 +1,3 @@
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-          package="com.mrousavy.camera">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
 
 </manifest>

